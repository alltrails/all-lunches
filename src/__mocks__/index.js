export { default as restaurantMock } from './restaurant.mock';
export { default as restaurantsMock } from './restaurants.mock';
export { default as reactRefMock } from './reactRef.mock';
