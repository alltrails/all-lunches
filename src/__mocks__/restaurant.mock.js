/* eslint-disable sort-keys  */

export default {
  coordinates: [-122.408999, 37.7902566],
  id: 'ChIJkVLoONZ_j4ARx2vJ4umERnI',
  name: "Uncle Vito's Pizza",
  photoUrl:
    'https://maps.googleapis.com/maps/api/place/js/PhotoService.GetPhoto?1sAap_uEBdIFbWc8LjoO8hOGn1QTJBeq21XOCk6ZHFGBmGWWqrImCdbERWb-K4T9Mq_QB5Mspi1Kiyc_GTU_XRbcReDLyGaIWdJWcssnsj3mZmy5xQsGCPeuOiBjjDEciT92r5iUeEAvSNn4dPN2r9rng9SH3fTXpxlT6BA5TB1MPzYfQ_GU5J&3u4032&5m1&2e1&callback=none&key=AIzaSyDg0Zd8r5Jf5i4MWlM9_eGYrpyRUeqBbPY&token=10623',
  placeUrl:
    "https://www.google.com/maps/search/?api=1&query=Uncle%20Vito's%20Pizza&query_place_id=ChIJkVLoONZ_j4ARx2vJ4umERnI",
  priceLevel: 1,
  rating: 4.3,
  supportingText: 'restaurant',
  userRatingsTotal: 1427,
};
