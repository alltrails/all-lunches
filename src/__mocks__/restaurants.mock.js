import DefaultStorePNG from 'assets/store-front.png';

export default [
  {
    coordinates: [-122.408999, 37.7902566],
    id: 'ChIJkVLoONZ_j4ARx2vJ4umERnI',
    name: "Uncle Vito's Pizza",
    photoUrl: DefaultStorePNG,
    placeUrl:
      "https://www.google.com/maps/search/?api=1&query=Uncle%20Vito's%20Pizza&query_place_id=ChIJkVLoONZ_j4ARx2vJ4umERnI",
    priceLevel: 1,
    rating: 4.3,
    supportingText: 'restaurant',
    userRatingsTotal: 1427,
  },
  {
    coordinates: [-122.408675, 37.7900735],
    id: 'ChIJg7cNIYyAhYAR8D4ml1oItTg',
    name: "Roxanne's Cafe",
    photoUrl: DefaultStorePNG,
    placeUrl:
      "https://www.google.com/maps/search/?api=1&query=Roxanne's%20Cafe&query_place_id=ChIJg7cNIYyAhYAR8D4ml1oItTg",
    priceLevel: 2,
    rating: 4.2,
    supportingText: 'restaurant',
    userRatingsTotal: 1219,
  },
  {
    coordinates: [-122.4073542, 37.7902509],
    id: 'ChIJaa9OAYyAhYARxwgUHLNEpOY',
    name: 'Bouche',
    photoUrl: DefaultStorePNG,
    placeUrl:
      'https://www.google.com/maps/search/?api=1&query=Bouche&query_place_id=ChIJaa9OAYyAhYARxwgUHLNEpOY',
    priceLevel: 3,
    rating: 4.5,
    supportingText: 'bar',
    userRatingsTotal: 465,
  },
  {
    coordinates: [-122.409088, 37.7902449],
    id: 'ChIJvXXIOoyAhYARe-ixq3WdmrY',
    name: 'Sons & Daughters',
    photoUrl: DefaultStorePNG,
    placeUrl:
      'https://www.google.com/maps/search/?api=1&query=Sons%20%26%20Daughters&query_place_id=ChIJvXXIOoyAhYARe-ixq3WdmrY',
    priceLevel: 4,
    rating: 4.6,
    supportingText: 'restaurant',
    userRatingsTotal: 249,
  },
];
