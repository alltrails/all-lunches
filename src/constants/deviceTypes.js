export const DESKTOP = 'desktop';
export const MOBILE = 'mobile';
export const TABLET = 'tablet';
