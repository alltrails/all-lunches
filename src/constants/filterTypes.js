export const HIGH_TO_LOW_FILTER_OPTION = 'high-to-low';
export const LOW_TO_HIGH_FILTER_OPTION = 'low-to-high';
