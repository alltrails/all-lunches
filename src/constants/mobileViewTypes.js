export const MAP_VIEW = 'map_view';
export const LIST_VIEW = 'list_view';
