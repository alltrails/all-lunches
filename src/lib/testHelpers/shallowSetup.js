import React from 'react';
import { shallow } from 'enzyme';

export default (Component, defaultProps) =>
  (otherProps = {}) => {
    const props = { ...defaultProps, ...otherProps };
    const wrapper = shallow(<Component {...props} />);

    return { props, wrapper };
  };
