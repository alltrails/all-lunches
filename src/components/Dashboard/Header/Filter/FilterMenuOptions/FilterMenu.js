import React from 'react';
import PropTypes from 'prop-types';

import { refType } from 'constants/propTypes';
import * as FilterTypes from 'constants/filterTypes';

import CheckmarkIcon from 'components/svg/CheckmarkIcon';
import FocusTrap from 'components/shared/FocusTrap';
import {
  ApplyButton,
  CheckmarkIconWrapper,
  Form,
  ListWrapper,
  OptionItem,
  OptionItemButton,
  Options,
} from './style';

const menuOptions = [
  { id: FilterTypes.HIGH_TO_LOW_FILTER_OPTION, title: 'Ratings High to Low' },
  { id: FilterTypes.LOW_TO_HIGH_FILTER_OPTION, title: 'Ratings Low to High' },
];

const FilterMenu = ({ menuRef, onMenuOptionClick, onSubmit, selectedMenuOption }) => (
  <FocusTrap fallbackFocus="#pop-out-menu">
    <Options ref={menuRef} tabIndex="-1" id="pop-out-menu">
      <Form>
        <ListWrapper>
          {menuOptions.map((option) => {
            const isSelected = selectedMenuOption === option.id;

            return (
              <OptionItemButton
                key={option.id}
                onClick={() => onMenuOptionClick(option.id)}
                type="button"
              >
                <OptionItem>
                  <CheckmarkIconWrapper isSelected={isSelected}>
                    <CheckmarkIcon
                      titleText={option.title}
                      variant={isSelected ? 'green' : 'white'}
                    />
                  </CheckmarkIconWrapper>
                  {option.title}
                </OptionItem>
              </OptionItemButton>
            );
          })}
        </ListWrapper>
        <ApplyButton disabled={!selectedMenuOption} onClick={onSubmit}>
          Apply
        </ApplyButton>
      </Form>
    </Options>
  </FocusTrap>
);
FilterMenu.propTypes = {
  menuRef: refType,
  onMenuOptionClick: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  selectedMenuOption: PropTypes.string,
};

FilterMenu.defaultProps = {
  menuRef: null,
  selectedMenuOption: null,
};

export default FilterMenu;
