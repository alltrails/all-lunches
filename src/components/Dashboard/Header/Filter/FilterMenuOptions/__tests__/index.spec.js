/* eslint-env jest */
/**
 * @jest-environment jsdom
 */

import shallowSetup from 'lib/testHelpers/shallowSetup';
import { withHooks } from 'jest-react-hooks-shallow';
import { reactRefMock } from '__mocks__';

import * as FilterTypes from 'constants/filterTypes';

import FilterMenuContainer from '../index';
import FilterMenu from '../FilterMenu';

describe('FilterMenuContainer', () => {
  const defaultProps = {
    filterButtonRef: reactRefMock,
    onClose: () => {},
    onMenuOptionClick: () => {},
    onSubmit: () => {},
    selectedMenuOption: null,
  };
  const setup = shallowSetup(FilterMenuContainer, defaultProps);

  it('renders the expected view', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders the expected view with a selected menu option', () => {
    const { wrapper } = setup({ selectedMenuOption: FilterTypes.HIGH_TO_LOW_FILTER_OPTION });
    expect(wrapper).toMatchSnapshot();
  });

  it('calls the onMenuOptionClick function prop with the selected menu option', () => {
    const { props, wrapper } = setup({ onMenuOptionClick: jest.fn() });

    const filterMenuComponent = wrapper.find(FilterMenu);
    filterMenuComponent.props().onMenuOptionClick();

    expect(props.onMenuOptionClick).toBeCalled();
  });

  it('calls the onSubmit function prop with the selected menu option', () => {
    const { props, wrapper } = setup({ onSubmit: jest.fn() });

    const filterMenuComponent = wrapper.find(FilterMenu);
    filterMenuComponent.props().onSubmit();

    expect(props.onSubmit).toBeCalled();
  });

  describe('Handling mousedown events', () => {
    const mousedownEvent = { target: {} };
    let eventListeners;

    beforeEach(() => {
      eventListeners = {};
      window.addEventListener = jest.fn((event, cb) => {
        eventListeners[event] = cb;
      });
      window.removeEventListener = jest.fn();
    });

    it('calls the onClose function prop when target is outside of menu or filter button', () => {
      withHooks(() => {
        const { props } = setup({ onClose: jest.fn() });
        eventListeners.mousedown(mousedownEvent);

        expect(props.onClose).toBeCalled();
      });
    });
  });
});
