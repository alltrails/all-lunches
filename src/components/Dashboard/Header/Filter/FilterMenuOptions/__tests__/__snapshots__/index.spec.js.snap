// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterMenuContainer renders the expected view 1`] = `
<FilterMenu
  menuRef={
    Object {
      "current": undefined,
    }
  }
  onMenuOptionClick={[Function]}
  onSubmit={[Function]}
  selectedMenuOption={null}
/>
`;

exports[`FilterMenuContainer renders the expected view with a selected menu option 1`] = `
<FilterMenu
  menuRef={
    Object {
      "current": undefined,
    }
  }
  onMenuOptionClick={[Function]}
  onSubmit={[Function]}
  selectedMenuOption="high-to-low"
/>
`;
